rudder-core/src/main/scala/com/normation/rudder/hooks/RunHooks.scala

```diff
@@ -0,0 +1,177 @@
+
+import org.slf4j.LoggerFactory
+import net.liftweb.common.Logger
+import net.liftweb.util.Helpers.tryo
+
+/*
+ * The goal of that file is to give a simple abstraction to run hooks in
+ * rudder.
+ *
+ * Hooks are stored in a directory. All hooks from a directory are
+ * run sequentially, so that side effects from one hook can be used
+ * in the following if the user want so.
+ * A hook wich fails stop the process and error from stderr are used
+ * for the reason of the failing.
+ * A failed hook is decided by the return code: 0 mean success, anything
+ * else is a failure.
+ *
+ * Hooks are asynchronously executed by default, in a Future.
+ */
+
+
+/*
+ * Hooks are group in "set". We run all the hooks
+ * from the same set with the same set of parameters.
+ * The hooks are executed in the order of the list.
+ */
+
+final case class Hooks(basePath: String, hooksFile: List[String])
+final case class HookParameters(values: List[String])
+
+/**
+ * Loggger for hooks
+ */
+object HooksLogger extends Logger {
+  override protected def _logger = LoggerFactory.getLogger("hooks")
+}
+
+object RunHooks {
+
+  /**
+   * Runs a list of hooks. Each hook is run sequencially, but
+   * the whole process is asynchronous.
+   * If one hook fails, the whole list fails.
+   */
+  def asyncRun(hooks: Hooks, parameters: HookParameters): Future[Box[Unit]] = {
+    /*
+     * We can not use Future.fold, because it execute all scripts
+     * in parallele and then combine their results. Our semantic
+     * is executre script one after the other, combining at each
+     * step.
+     * But we still want the whole operation to be non-bloking.
+     */
+    ( Future(Full(()):Box[Unit]) /: hooks.hooksFile) { case (previousFuture, nextHookName) =>
+      val path = hooks.basePath + File.separator + nextHookName
+      previousFuture.flatMap {
+        case Full(())     =>
+          HooksLogger.debug(s"Run hook: '${path} ${parameters.values.mkString(" ")}'")
+          RunNuCommand.run(Cmd(path, parameters.values)).map { result =>
+            lazy val msg = s"Exit code ${result.code} when executing '${path} ${parameters.values.mkString(" ")}'. \n  Stdout: '${result.stdout}' \n  Stderr: '${result.stderr}'"
+            HooksLogger.trace(s"  -> results: ${msg}")
+            if(result.code != 0) {
+              Failure(msg)
+            } else {
+              Full(())
+            }
+          } recover {
+            case ex: Exception => Failure(s"Exception when executing '${path} ${parameters.values.mkString(" ")}: ${ex.getMessage}")
+          }
+        case eb: EmptyBox => Future(eb)
+      }
+    }
+  }
+
+  /*
+   * Run hooks in given directory, synchronously.
+   *
+   * The hooks are sorted alpha-numerically, so that
+   * standard ordering like
+   * 10-first-hook
+   * 20-second-hook
+   * etc
+   * works as expected.
+   *
+   * File starting with "_" are ignored
+   *
+   */
+  def syncRun(hooks: Hooks, parameters: HookParameters): Box[Unit] = {
```

Comment:
Ha, the comment was not updated when I changed the implementation. Will correct it.